packages/headless-components/cms/src/react/CmsCollection.tsx

```diff
@@ -70,53 +75,252 @@ export const Root = React.forwardRef<HTMLDivElement, RootProps>(
+ * </CmsCollection.Items>
+ *
+ * // Using asChild pattern
+ * <CmsCollection.Items asChild emptyState={<div>No data available</div>}>
+ *   <section className="grid grid-cols-3 gap-4">
+ *     <CmsCollection.ItemRepeater>
+ *       <CmsItem.Field fieldId="title" />
+ *       <CmsItem.Field fieldId="description" />
+ *     </CmsCollection.ItemRepeater>
+ *   </section>
+ * </CmsCollection.Items>
+ * ```
+ */
+export const Items = React.forwardRef<HTMLElement, ItemsProps>((props, ref) => {
+  const { children, emptyState, asChild, className, ...otherProps } = props;
+
+  return (
+    <CoreCmsCollection.Items>
+      {({ items, isLoading, error }) => {
+        // Don't render anything during loading or error states
+        if (isLoading || error) {
+          return null;
+        }
+
+        // Show empty state when no items
+        if (items.length === 0) {
+          return emptyState || null;
+        }
+
+        const dataAttributes = {
+          'data-testid': TestIds.cmsCollectionItems,
+          'data-empty': items.length === 0,
+        };
+
+        if (asChild) {
+          return (
+            <AsChildSlot
+              ref={ref}
+              asChild={asChild}
+              className={className}
+              {...dataAttributes}
+              {...otherProps}
+            >
+              {children}
+            </AsChildSlot>
+          );
+        }
+
+        return (
+          <div
+            {...dataAttributes}
+            className={className}
+            ref={ref as React.LegacyRef<HTMLDivElement>}
+            {...otherProps}
+          >
+            {children}
+          </div>
+        );
+      }}
+    </CoreCmsCollection.Items>
+  );
+});
+
+/**
+ * Props for CmsCollection.ItemRepeater component
+ */
+export interface ItemRepeaterProps {
+  children: React.ReactNode;
+  asChild?: boolean;
+  className?: string;
+}
+
+/**
+ * Repeats for each collection item in the list, providing individual item context.
+ *
+ * @component
+ * @example
+ * ```tsx
+ * <CmsCollection.ItemRepeater className="item-card">
+ *   <CmsItem.Field fieldId="title" />
+ *   <CmsItem.Field fieldId="image" />
+ *   <CmsItem.Field fieldId="description" />
+ * </CmsCollection.ItemRepeater>
+ * ```
+ */
+export function ItemRepeater(props: ItemRepeaterProps) {
+  const { children, asChild, className, ...otherProps } = props;
+
+  return (
+    <CoreCmsCollection.ItemRepeater>
+      {({ items, collectionId, isLoading, error }) => {
+        // Don't render during loading or error states
+        if (isLoading || error || items.length === 0) {
+          return null;
+        }
+
+        return (
+          <>
+            {items.map((item) => (
+              <CmsItem.Root
+                key={item._id}
+                item={{
+                  collectionId,
+                  id: item._id,
+                  item: item,
+                }}
+              >
+                {asChild ? (
+                  <AsChildSlot
```

Comment:
You don't have to check the asChild prop value if you are using the AsChildSlot
same as the comment above